### ISSUE TITLE
"brand-ft-masthead" is still the old logo from before the paper redesign

### ISSUE THREAD

**Issue by callumlocke** [Financial-Times/o-ft-icons#91]
old logo: http://image.webservices.ft.com/v1/images/raw/fticon:brand-ft-masthead?source=x

new logo looks like this: http://ig.ft.com/sites/2015/boko-haram-timeline/

**Comment by sdbernard** [comment]
+1 for getting this changed asap

**Comment by alicebartlett** [comment]
Oooft. Is there an SVG of the new logo somewhere @danskinner?

**Comment by callumlocke** [comment]
This is the one I've been using:

```svg
<svg xmlns="http://www.w3.org/2000/svg" width="843" height="68" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 843 68">
  <path d="M20.2 58c0 5 1.3 6 10 6v3H0v-3c6 0 8-.6 8-5.8V9.8C8 4.6 6 4 0 4V1h41.6C47 1 49 1 50.8.7L51 19h-2.4C47.2 5.7 44 4.5 31.3 4.5h-11v27h9c8.5 0 9-1.5 9.7-8.2h2.4V43H39c-.7-6.8-1.2-8-9.8-8h-9v23M59 67v-3c6 0 8-.6 8-5.8V9.8C67 4.6 65 4 59 4V1h28.3v3c-6 0-8 .6-8 5.8v48.4c0 5.2 2 5.8 8 5.8v3H59m94.7 1l-45-55.8V58c0 5.2 3.8 6 10 6v3h-24v-3c6.2 0 9.7-.8 9.7-6V7c-3-3-5-3-9.6-3V1h21l34.6 44.3V10c0-5.2-3.6-6-9.8-6V1h24v3c-6.3 0-9.8.8-9.8 6v58h-1m29.8-1h-23.8v-3c6.3 0 9.2-1 11-6.2L192.5.6h6l23 57.6c2.2 5 3 5.8 7.8 5.8v3h-27.6v-3c8.5 0 9.3-.7 7.2-5.8l-7-17.4h-21l-6 17c-2 5 1 6.2 8 6.2v3m-1-29.8h18l-9-23.6-9 23.6z"/>
  <defs>
    <path id="a" d="M0 0h842.2v68H0z"/>
  </defs>
  <clipPath id="b">
    <use xlink:href="#a" overflow="visible"/>
  </clipPath>
  <path d="M291 68l-44.8-55.8V58c0 5.2 3.6 6 9.8 6v3h-23.3v-3c5.7 0 9-.8 9-6V7.2c-3.2-3-5.6-3.2-11-3.2V1h22.6l34.5 44.5V10c0-5.2-3.7-6-9.8-6V1h23.8v3c-6 0-9.6.8-9.6 6v58h-1M356 1h1.5l1 19.7H356c-1.7-11.2-7.6-17-18.6-17C325.2 3.7 316 14 316 30c0 20.6 12.8 32.2 26.4 32.2 6 0 11.2-1.6 16.3-7l1.8 1.8c-4.2 6-12.5 11-23.6 11-17 0-34-12.5-34-33.4C303 14.4 318 0 337 0c8.8 0 13.5 3.8 15.8 3.8 1.3 0 2-1 3-2.8" clip-path="url(#b)"/>
  <path d="M366 67v-3c6 0 8-.6 8-5.8V9.8c0-5.2-2-5.7-8-5.7V1h28.2v3c-6 0-8 .6-8 5.8v48.4c0 5.2 2 5.8 8 5.8v3H366m55 0h-22.3v-3c5 0 7.5-1 9.5-6.2L429.7.6h6l23 57.6c2 5 3 5.8 7.8 5.8v3H439v-3c8.4 0 9-.7 7-5.8l-6.8-17.4h-20.6l-6 17c-2 5 1 6.2 8.3 6.2v3m-1-30h17l-9-23.6-8 23.6zm51 30v-3c6 0 8-.6 8-5.8V9.8c0-5.2-2-5.7-8-5.7V1h28v3c-6 0-8 .6-8 5.8v49c0 4.2 2 4.8 6 4.8h3c12 0 17-2 20-16.4l2 .3-2 19.4h-50M607 0l.3 19.6h-2.5C603.4 6 600 4.3 587.5 4.3h-4V58c0 5.2 2 6 10 6v3H561v-3c8.3 0 10.4-.8 10.4-6V4.5h-4c-12.6 0-16 1.7-17.4 15.8h-2.6l.3-19.6c1.8.3 3.7.5 9.3.5h40.8c5.5 0 7.5 0 9.2-.4m7.2 66.2v-3c6 0 8-.6 8-5.8V9.8c0-5.2-2-5.7-8-5.7V1h28.3v3c-6 0-8 .6-8 5.8v48.4c0 5.2 2 5.8 8 5.8v3h-28.3m113-66v3c-6 0-8.4.4-8 5.6l5 48.8c.8 5 2.7 5.6 8.6 5.6v3h-28v-3c5.7 0 7.8-.6 7.3-5.6L707 7l-21 61h-.8l-20-61-5 51.3c-.6 5.3 3 5.7 9 5.7v3h-22v-3c6 0 8-1 8.5-5.7l5-48.7c.5-5.2-2-5.5-8-5.5V1h23l14 45 15-45h22.6"/>
  <defs>
    <path id="c" d="M0 0h842.2v68H0z"/>
  </defs>
  <clipPath id="d">
    <use xlink:href="#c" overflow="visible"/>
  </clipPath>
  <path d="M780 42.8h-2.5c-.7-6.7-1.2-8-9.8-8h-10v24c0 4.2 2.2 4.8 5.8 4.8h5.4c12 0 16-2 20-16.4l2 .3-2 19.5h-52v-3c6 0 8-.8 8-6V10c0-5.2-2-5.7-8-5.7V1h50v16.5h-2c-2-11.5-5-13-18-13h-10v26.8h10c8.6 0 9.2-1.4 10-8h2.2v19.5m52 10.5c0-6-4-9-11-12.5l-10-5.5c-7.8-4.2-12.5-8.8-12.5-17C799 8 807.5 0 819.2 0c8 0 12 3.8 14 3.8 1.7 0 2.4-1 3-2.8h2l1 18.5h-2.5C835.5 10.8 829 3.8 820 3.8c-6.7 0-11.2 4-11.2 9.8 0 6.3 4.7 8.8 10 11.7l9 4.8c8.4 5 14.4 10 14.4 18 0 12-10 20-22.6 20-8.8 0-13-4-15.5-4-1 0-2 2-2 4h-2l-1-20h2.7c1.8 12 10.3 17 18.8 17 7 0 13-3 13-11" clip-path="url(#d)"/>
</svg>
```

It's nice and optimised but it's got a little glitch on the "S" which is visible when you use it at large sizes. (You can paste it into [svgomg](https://jakearchibald.github.io/svgomg/) to see what I mean.) I've only been using it small.

@danskinner may have a better one.

**Comment by sdbernard** [comment]
probably best to get the original from Kevin and make an svg from that

**Comment by alicebartlett** [comment]
@sdbernard @callumlocke OK thanks I'll get the original from Kevin

**Comment by alicebartlett** [comment]
#92

**Comment by danskinner** [comment]
@alicebartlett Did you get hold of it? I have an eps..

**Comment by alicebartlett** [comment]
Yes thanks @danskinner, Kevin sent me a pdf. The PR has been merged, we just need to create a new release, but it will be a breaking change as the padding around the masthead as included in the SVG has changed.

**Comment by kavanagh** [comment]
Did this release just break stuff? Getting lots of broken icons.

Looks like the codepoints in that PR changed.

cc @sdbernard

**Comment by AlbertoElias** [comment]
Code points change every time the icon font is regenerated. But if you bring in the latest version of icons, you should also be getting the latest icon font

**Comment by kavanagh** [comment]
Why should we have to upgrade? That's a lot of work for us. The build service is not much good if things that used to work break. If this is the case we'll stop using this module now.

**Comment by AlbertoElias** [comment]
Oh, I'm not saying it's not an issue. What I mean is, "if code points change, icon fonts change". They both go together. It's weird that you're getting the latest code point changes, but not the new icon font. Could you send me an example url of where this is failing? Might be a cache issue.

**Comment by AlbertoElias** [comment]
So I've been checking out how you're using icons, and I believe the error comes because you're installing it via bower, but requesting the icon font from the build service.

If you install icons ~v2.1, which resolves to v2.1.5 for example, which has an icon font with hash `12345`.

But using o-assets you point to the build service, version ^v2.1, which resolves to v2.4.5, which has an icon font with hash `54321`.

So here, o-assets is going to take the build service url which is going to resolve to v2.4.5, and search there for the hash `12345` because that's the version installed via bower, and you're compiling via Sass.

I hope that makes sense

**Comment by alicebartlett** [comment]
Going to close this issue now.

@kavanagh and I had a chat yesterday and it seems like the problem with Rugby World Cup and Elections was an issue introduced by this minor release: https://github.com/Financial-Times/o-ft-icons/releases/tag/v2.4.0, not the major one.

@kavanagh give us a shout if you need any help with squaring the build-service o-icons vs the bower version that @AlbertoElias is talking about ^up there, and if you have any thoughts on how we could better prevent others from falling into this trap in the future I'd be interested in hearing them.

And just to be 100% clear on this:

> Why should we have to upgrade? That's a lot of work for us. The build service is not much good if things that used to work break. If this is the case we'll stop using this module now.

Origami should never be breaking components on minor releases, if it does, that's by mistake not on purpose. Forcing people to upgrade was also not what @AlbertoElias was suggesting.

**Comment by kavanagh** [comment]
I this chatted through @AlbertoElias, thanks to him I understand the finer points of this now. I've decided I'll use the build service exclusively for icons now, at least until a stable SVG solution lands.

@AlbertoElias and my understanding is that:

* the build script generates a new code point for each icon during each release/build.
* this makes it necessary to hash the file names of the built icon font, whereas file names of assets, such as images and fonts, don't usually get hashed in other Origami modules.
* you cannot mix the use of the build service and the bower component (via o-assets) unless you fix on a specific version, for example `2.3.7`. Ranges like `^2` or `~2.3.7` wont work because subsequent release will have a new hashed/rev'd file name. This isn't a problem with other origami modules with assets (such as o-share) because the files names are not hashed.

So you cant do this:

```scss
// my-styles.scss
@include oAssetsSetModulePaths((
  // use caret range
  'o-ft-icons': '//build.origami.ft.com/files/o-ft-icons@^2'
));
```

Or even this:

```scss
// my-styles.scss
@include oAssetsSetModulePaths((
  // use tilda range
  'o-ft-icons': '//build.origami.ft.com/files/o-ft-icons@~2.3.7'
));
```

-----

So you have the following options:

1. use the build service exclusively, which means you get all that css that your application doesn't need
2. you don't use the build service at all (only the bower module) and serve the font files yourself
3. you mix bower and build service, using o-assets to set a **specific** version of a font as below:

```scss
// my-styles.scss
@include oAssetsSetModulePaths((
  'o-ft-icons': '//build.origami.ft.com/files/o-ft-icons@2.3.7'
));
```

The downside to options 2 and 3 is that you don't get automatic updates to the font files when fixes or new icons become available.

**Comment by AlbertoElias** [comment]
Thanks a lot for the write up! Great comment to point to if this occurs again
